consensus/src/traits.rs

```diff
@@ -153,36 +152,85 @@ pub trait ConsensusApi<
         .await;
     }
 
-    /// Signs a vote
-    fn sign_vote(
+    /// Sign a DA proposal.
+    fn sign_da_proposal(
         &self,
-        leaf_commitment: &Commitment<LEAF>,
-        _view_number: TYPES::Time,
-    ) -> (EncodedPublicKey, EncodedSignature) {
-        let signature = TYPES::SignatureKey::sign(self.private_key(), leaf_commitment.as_ref());
-        (self.public_key().to_bytes(), signature)
+        block_commitment: &Commitment<TYPES::BlockType>,
+    ) -> EncodedSignature {
+        let signature = TYPES::SignatureKey::sign(self.private_key(), block_commitment.as_ref());
+        signature
     }
 
-    /// Signs a proposal
-    fn sign_proposal(
+    /// Sign a validating or commitment proposal.
+    fn sign_validating_or_commitment_proposal(
         &self,
         leaf_commitment: &Commitment<LEAF>,
-        _view_number: TYPES::Time,
     ) -> EncodedSignature {
         let signature = TYPES::SignatureKey::sign(self.private_key(), leaf_commitment.as_ref());
         signature
     }
 
-    /// Validate a quorum certificate by checking
-    /// signatures
-    fn validate_qc(&self, quorum_certificate: &LEAF::QuorumCertificate) -> bool;
+    /// Sign a vote on DA proposal.
+    fn sign_da_vote(
+        &self,
+        block_commitment: Commitment<TYPES::BlockType>,
+    ) -> (EncodedPublicKey, EncodedSignature) {
+        let signature = TYPES::SignatureKey::sign(
+            self.private_key(),
+            &VoteData::<TYPES, LEAF>::DA(block_commitment).as_bytes(),
+        );
+        (self.public_key().to_bytes(), signature)
+    }
+
+    /// Sign a positive vote on validating or commitment proposal.
+    fn sign_yes_vote(
+        &self,
+        leaf_commitment: Commitment<LEAF>,
+    ) -> (EncodedPublicKey, EncodedSignature) {
+        let signature = TYPES::SignatureKey::sign(
+            self.private_key(),
+            &VoteData::<TYPES, LEAF>::Yes(leaf_commitment).as_bytes(),
```

Comment:
It may be worth adding a comment here explaining why this is correct. IMO this is a bit subtle. Something like:
```
/// the leaf commitment and the type of the vote (yes) are signed
/// This is the minimum amount of information necessary. 
/// Any users of the subsequently constructed QC should be able to check 
/// that this node voted `Yes` on that leaf. 
/// The leaf is expected to be reconstructed based on other information in the yes vote.
```

Reply:
Sounds good! I combined your suggested changes for `sign_yes_vote` and `sign_timeout_vote` and added comments for all vote signing functions.

Reply:
🎉